fleetshard/pkg/central/operator/upgrade.go

```diff
@@ -4,46 +4,56 @@ package operator
 import (
 	"context"
 	"fmt"
-	"strings"
-
+	containerImage "github.com/containers/image/docker/reference"
 	"github.com/stackrox/acs-fleet-manager/fleetshard/pkg/central/charts"
 	"golang.org/x/exp/slices"
 	"helm.sh/helm/v3/pkg/chart"
 	"helm.sh/helm/v3/pkg/chartutil"
 	appsv1 "k8s.io/api/apps/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/validation"
 	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	operatorNamespace        = "rhacs"
 	releaseName              = "rhacs-operator"
-	operatorDeploymentPrefix = "rhacs-operator-manager"
+	operatorDeploymentPrefix = "rhacs-operator"
 
-	// deployment names should contain at most 63 characters
 	// RFC 1035 Label Names: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names
-	maxOperatorDeploymentNameLength = 63
+	maxLabelLength = 63
 )
 
-func parseOperatorImages(images []string) ([]chartutil.Values, error) {
+// DeploymentConfig represents operator configuration for deployment
+type DeploymentConfig struct {
+	Image  string
+	GitRef string
+}
+
+func parseOperatorConfigs(operators []DeploymentConfig) ([]chartutil.Values, error) {
 	var operatorImages []chartutil.Values
 	uniqueImages := make(map[string]bool)
-	for _, img := range images {
-		if !strings.Contains(img, ":") {
-			return nil, fmt.Errorf("failed to parse image %q", img)
+	for _, operator := range operators {
+		imageReference, err := containerImage.Parse(operator.Image)
+		if err != nil {
+			return nil, err
 		}
-		strs := strings.Split(img, ":")
-		if len(strs) != 2 {
-			return nil, fmt.Errorf("failed to split image and tag from %q", img)
+		image := imageReference.String()
+		errs := validation.IsValidLabelValue(operator.GitRef)
+		if errs != nil {
```

Comment:
minor: this could be inlined
```suggestion
		if errs := validation.IsValidLabelValue(operator.GitRef); errs != nil {
```